components/datetime/src/format/neo.rs

```diff
@@ -855,53 +874,78 @@ impl RawDateTimeNames {
         self.week_calculator = Some(week_calculator);
     }
 
+    pub(crate) fn load_fixed_decimal_formatter(
+        &mut self,
+        loader: &impl FixedDecimalFormatterLoader,
+        locale: &DataLocale,
+    ) -> Result<(), Error> {
+        let mut options = FixedDecimalFormatterOptions::default();
+        options.grouping_strategy = GroupingStrategy::Never;
+        self.fixed_decimal_formatter =
+            Some(FixedDecimalFormatterLoader::load(loader, locale, options)?);
+        Ok(())
+    }
+
     /// Associates this [`TypedDateTimeNames`] with a pattern
     /// without loading additional data for that pattern.
     #[inline]
     pub(crate) fn with_pattern<'l>(
         &'l self,
-        pattern: &'l Pattern,
+        pattern: DateTimePatternBorrowed<'l>,
     ) -> RawDateTimePatternFormatter<'l> {
         RawDateTimePatternFormatter {
             pattern,
             names: self.as_borrowed(),
         }
     }
 
-    pub(crate) fn load_for_pattern<'l, YearMarker, MonthMarker>(
-        &'l mut self,
-        year_provider: &(impl DataProvider<YearMarker> + ?Sized),
-        month_provider: &(impl DataProvider<MonthMarker> + ?Sized),
-        weekday_provider: &(impl DataProvider<WeekdayNamesV1Marker> + ?Sized),
-        dayperiod_provider: &(impl DataProvider<DayPeriodNamesV1Marker> + ?Sized),
-        pattern: &'l Pattern,
-        week_calculator_loader: impl FnOnce(
-            &DataLocale,
-        )
-            -> Result<WeekCalculator, icu_calendar::CalendarError>,
-    ) -> Result<RawDateTimePatternFormatter<'l>, Error>
+    /// Loads all data required for formatting the given [`PatternItem`]s.
+    ///
+    /// This function has a lot of arguments because many of the arguments are generic,
+    /// and pulling them out to an options struct would be cumbersome.
+    #[allow(clippy::too_many_arguments)]
+    pub(crate) fn load_for_pattern<YearMarker, MonthMarker>(
+        &mut self,
+        year_provider: Option<&(impl DataProvider<YearMarker> + ?Sized)>,
```

Comment:
Hmm, with AnyCalendar `<YearMarker, MonthMarker>` are not known until runtime and I don't really want to include more than 1 copy of this function (i.e. through an enum switch). Maybe we can move those over to the external loader architecture so that the enum switch is isolated to the load function. Let's figure that out when I implement the AnyCalendar version of neo formatter.

Reply:
Yep.